src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -2357,3 +2365,1577 @@ IsForeignTable(Oid relationId)
+		Datum values[Natts_pg_dist_background_tasks] = { 0 };
+		bool nulls[Natts_pg_dist_background_tasks] = { 0 };
+
+		memset(nulls, true, sizeof(nulls));
+
+		int64 taskId = GetNextBackgroundTaskTaskId();
+
+		values[Anum_pg_dist_background_tasks_job_id - 1] = Int64GetDatum(jobId);
+		nulls[Anum_pg_dist_background_tasks_job_id - 1] = false;
+
+		values[Anum_pg_dist_background_tasks_task_id - 1] = Int64GetDatum(taskId);
+		nulls[Anum_pg_dist_background_tasks_task_id - 1] = false;
+
+		values[Anum_pg_dist_background_tasks_owner - 1] = ObjectIdGetDatum(owner);
+		nulls[Anum_pg_dist_background_tasks_owner - 1] = false;
+
+		Oid statusOid = InvalidOid;
+		if (dependingTaskCount == 0)
+		{
+			statusOid = CitusTaskStatusRunnableId();
+		}
+		else
+		{
+			statusOid = CitusTaskStatusBlockedId();
+		}
+		values[Anum_pg_dist_background_tasks_status - 1] = ObjectIdGetDatum(statusOid);
+		nulls[Anum_pg_dist_background_tasks_status - 1] = false;
+
+		values[Anum_pg_dist_background_tasks_command - 1] = CStringGetTextDatum(command);
+		nulls[Anum_pg_dist_background_tasks_command - 1] = false;
+
+		HeapTuple newTuple = heap_form_tuple(RelationGetDescr(pgDistBackgroundTasks),
+											 values, nulls);
+		CatalogTupleInsert(pgDistBackgroundTasks, newTuple);
+
+		task = palloc0(sizeof(BackgroundTask));
+		task->taskid = taskId;
+		task->status = BACKGROUND_TASK_STATUS_RUNNABLE;
+		task->command = pstrdup(command);
+	}
+
+	/* 3. insert dependencies into catalog */
+	{
+		for (int i = 0; i < dependingTaskCount; i++)
+		{
+			Assert(pgDistbackgroundTasksDepend != NULL);
+
+			Datum values[Natts_pg_dist_background_tasks_depend] = { 0 };
+			bool nulls[Natts_pg_dist_background_tasks_depend] = { 0 };
+			memset(nulls, true, sizeof(nulls));
+
+			values[Anum_pg_dist_background_tasks_depend_job_id - 1] =
+				Int64GetDatum(jobId);
+			nulls[Anum_pg_dist_background_tasks_depend_job_id - 1] = false;
+
+			values[Anum_pg_dist_background_tasks_depend_task_id - 1] =
+				Int64GetDatum(task->taskid);
+			nulls[Anum_pg_dist_background_tasks_depend_task_id - 1] = false;
+
+			values[Anum_pg_dist_background_tasks_depend_depends_on - 1] =
+				Int64GetDatum(dependingTaskIds[i]);
+			nulls[Anum_pg_dist_background_tasks_depend_depends_on - 1] = false;
+
+			HeapTuple newTuple = heap_form_tuple(
+				RelationGetDescr(pgDistbackgroundTasksDepend), values, nulls);
+			CatalogTupleInsert(pgDistbackgroundTasksDepend, newTuple);
+		}
+	}
+
+	if (pgDistbackgroundTasksDepend)
+	{
+		table_close(pgDistbackgroundTasksDepend, NoLock);
+	}
+	table_close(pgDistBackgroundTasks, NoLock);
+
+	return task;
+}
+
+
+/*
+ * ResetRunningBackgroundTasks finds all tasks currently in Running state and resets their
+ * state back to runnable.
+ *
+ * Tasks that were running during a failover, crash or restart will still be marked
+ * running after. This function assumes appropriate care by the caller that the task is in
+ * fact _not_ running at the moment on the primary coordinator of the formation.
+ *
```

Comment:
I think this comment is a outdated, because the code suggests that it doesn't matter if the task was running or not. If it's running this call will cancel it.

Reply:
yes, comment was old.